pypeit/flatfield.py

```diff
@@ -903,74 +907,132 @@ def fit(self, debug=False):
                 ax.legend()
                 ax.set_xlabel('Normalized Slit Position')
                 ax.set_ylabel('Residuals from pixelflat 2-d fit')
-                ax.set_title('Spatial Residuals for slit={:d}'.format(slit))
+                ax.set_title('Spatial Residuals for slit={:d}'.format(slit_spat))
                 plt.show()
 
             # Save the 2D residual model
             twod_model[...] = 1.
             if exit_status > 1:
                 msgs.warn('Two-dimensional fit to flat-field data failed!  No higher order '
-                          'flat-field corrections included in model of slit {0}!'.format(slit))
+                          'flat-field corrections included in model of slit {0}!'.format(slit_spat))
             else:
                 twod_model[twod_gpm] = twod_flat_fit[np.argsort(twod_srt)]
 
             # Construct the full flat-field model
             # TODO: Why is the 0.05 here for the illumflat compared to the 0.01 above?
-            self.flat_model[onslit] = twod_model[onslit] \
-                                        * np.fmax(self.msillumflat[onslit], 0.05) \
-                                        * np.fmax(spec_model[onslit], 1.0)
+            self.flat_model[onslit_tweak] = twod_model[onslit_tweak] \
+                                        * np.fmax(self.msillumflat[onslit_tweak], 0.05) \
+                                        * np.fmax(spec_model[onslit_tweak], 1.0)
 
             # Construct the pixel flat
-            self.mspixelflat[onslit] = rawflat[onslit]/self.flat_model[onslit]
+            # TODO -- JFH :: Are we ok with this change??
+            #self.mspixelflat[onslit] = rawflat[onslit]/self.flat_model[onslit]
+            self.mspixelflat[onslit_tweak] = 1.
+            self.mspixelflat[onslit_trimmed] = rawflat[onslit_trimmed]/self.flat_model[onslit_trimmed]
```

Comment:
Like this:

```
            self.mspixelflat[onslit_tweak] = 1.
            trimmed_slitid_img_anew = self.slits.slit_img(pad=-trim)
            onslit_trimmed_anew = trimmed_slitid_img_anew == slit_spat
            self.mspixelflat[onslit_trimmed_anew] = rawflat[onslit_trimmed_anew]/self.flat_model[onslit_trimmed_anew]
```

Reply:
Yes this is correct, but I think you just want to make the slit image for the slit in question not all the slits, which I think is an option to slit_img. Since we are in a loop here over the slits, right?

Reply:
Please paste in what you mean.